com.unity.render-pipelines.high-definition/Editor/Material/ShaderGraph/Nodes/EmissionNode.cs

```diff
@@ -116,17 +110,17 @@ public void GenerateNodeCode(ShaderStringBuilder sb, GenerationMode generationMo
 
         string GetFunctionName()
         {
-            return $"Unity_HDRP_GetEmissionHDRColor_{concretePrecision.ToShaderString()}";
+            return "Unity_HDRP_GetEmissionHDRColor_$precision";
```

Comment:
Changes like this are to meet the requirements of declaring a generic precision-switching function -- if it is generic, you must have the tokens in both your function name and your actual function code, and non-generic functions should not have "$" anywhere.  We now validate this in ProvideFunction().  This ensures we can correctly differentiate between precision-switched functions and plain old functions and treat them appropriately.  This validation also caught a number of issues in existing nodes.

Reply:
It actually does this validation all the time now (it's in FunctionRegistry.ProvideFunction).

I did discover a number of issues with multiple instances of a node at different precisions / different input types causing name collisions though.  So at some point we probably want to add some tests that try to instantiate all the variants of a node in a single graph to check for that.  This will be a lot easier once we have the graph manipulation and shader sandbox stuff though.